core/net/rpl/rpl-dag.c

```diff
@@ -1100,6 +1107,19 @@ rpl_recalculate_ranks(void)
     }
     p = nbr_table_next(rpl_parents, p);
   }
+  if (!updated) {
```

Comment:
The PR and its commits are quite old, I believe this code was necessary when rpl_parents table did not exist and only the current dag parents were updated.